File: backend/src/models/DeviceGuide.js
const { DataTypes } = require('sequelize');
const sequelize = require('../config/database');

const DeviceGuide = sequelize.define('DeviceGuide', {
  id: {
    type: DataTypes.INTEGER,
    primaryKey: true,
    autoIncrement: true,
  },
  name: {
    type: DataTypes.STRING(100),
    allowNull: false,
  },
  slug: {
    type: DataTypes.STRING(100),
    allowNull: true,
  },
  subtitle: {
    type: DataTypes.STRING(200),
    defaultValue: '',
  },
  icon: {
    type: DataTypes.STRING(100),
    defaultValue: 'setting-o',
  },
  iconUrl: {
    type: DataTypes.STRING(500),
    defaultValue: '',
  },
  iconUrlThumb: {
    type: DataTypes.STRING(500),
    defaultValue: '',
    comment: '图标缩略图URL，可选',
  },
  emoji: {
    type: DataTypes.STRING(20),
    defaultValue: '',
  },
  gradient: {
    type: DataTypes.STRING(300),
    defaultValue: 'linear-gradient(135deg, #3B82F6, #1D4ED8)',
  },
  badge: {
    type: DataTypes.STRING(20),
    defaultValue: '',
  },
  tags: {
    type: DataTypes.TEXT,
    allowNull: true,
    get() {
      const raw = this.getDataValue('tags');
      try { return JSON.parse(raw || '[]'); } catch { return []; }
    },
    set(val) {
      this.setDataValue('tags', JSON.stringify(val || []));
    },
  },
  sections: {
    type: DataTypes.TEXT('long'),
    allowNull: true,
    get() {
      const raw = this.getDataValue('sections');
      try { return JSON.parse(raw || '[]'); } catch { return []; }
    },
    set(val) {
      this.setDataValue('sections', JSON.stringify(val || []));
    },
  },
  coverImage: {
    type: DataTypes.STRING(500),
    defaultValue: '',
  },
  coverImageThumb: {
    type: DataTypes.STRING(500),
    defaultValue: '',
    comment: '封面缩略图URL，可选',
  },
  showcaseVideo: {
    type: DataTypes.STRING(500),
    defaultValue: '',
  },
  description: {
    type: DataTypes.TEXT,
    allowNull: true,
  },
  mediaItems: {
    type: DataTypes.TEXT('long'),
    allowNull: true,
    get() {
      const raw = this.getDataValue('mediaItems');
      try { return JSON.parse(raw || '[]'); } catch { return []; }
    },
    set(val) {
      this.setDataValue('mediaItems', JSON.stringify(val || []));
    },
  },
  helpItems: {
    type: DataTypes.TEXT('long'),
    allowNull: true,
    get() {
      const raw = this.getDataValue('helpItems');
      try { return JSON.parse(raw || '[]'); } catch { return []; }
    },
    set(val) {
      this.setDataValue('helpItems', JSON.stringify(val || []));
    },
  },
  categoryId: {
    type: DataTypes.INTEGER,
    allowNull: true,
    comment: '所属商品种类 ID',
  },
  sortOrder: {
    type: DataTypes.INTEGER,
    defaultValue: 0,
  },
  qrcodeUrl: {
    type: DataTypes.STRING(500),
    defaultValue: '',
  },
  status: {
    type: DataTypes.ENUM('active', 'inactive'),
    defaultValue: 'active',
  },
}, {
  tableName: 'device_guides',
  timestamps: true,
  indexes: [
    { unique: true, fields: ['slug'], name: 'slug' },
  ],
});

module.exports = DeviceGuide;

File: backend/internal/models/deviceguide.go
package models

import (
	"time"

	"gorm.io/datatypes"
)

type DeviceGuide struct {
	ID                 int            `gorm:"primaryKey" json:"id"`
	Name               string         `gorm:"size:100;not null" json:"name"`
	Slug               *string        `gorm:"size:100;uniqueIndex:slug" json:"slug"`
	Subtitle           string         `gorm:"size:200" json:"-"`
	Icon               string         `gorm:"size:100" json:"icon"`
	IconURL            string         `gorm:"column:iconUrl;size:500" json:"iconUrl"`
	IconURLThumb       string         `gorm:"column:iconUrlThumb;size:500" json:"iconUrlThumb"`
	Emoji              string         `gorm:"size:20" json:"emoji"`
	Gradient           string         `gorm:"size:300" json:"gradient"`
	Badge              string         `gorm:"size:20" json:"badge"`
	Tags               datatypes.JSON `gorm:"type:text" json:"tags"`
	Sections           datatypes.JSON `gorm:"type:longtext" json:"sections"`
	CoverImage         string         `gorm:"column:coverImage;size:500" json:"coverImage"`
	CoverImageThumb    string         `gorm:"column:coverImageThumb;size:500" json:"coverImageThumb"`
	ShowcaseVideo      string         `gorm:"column:showcaseVideo;size:500" json:"showcaseVideo"`
	Description        string         `gorm:"type:text" json:"description"`
	MediaItems         datatypes.JSON `gorm:"type:longtext;column:mediaItems" json:"mediaItems"`
	HelpItems          datatypes.JSON `gorm:"type:longtext;column:helpItems" json:"helpItems"`
	ManualPdfURL       string         `gorm:"column:manualPdfUrl;size:500" json:"manualPdfUrl"`
	CategoryID         *int           `gorm:"column:categoryId" json:"categoryId"`
	SortOrder          int            `gorm:"column:sortOrder" json:"sortOrder"`
	QrcodeURL          string         `gorm:"column:qrcodeUrl;size:500" json:"qrcodeUrl"`
	NameEn             string         `gorm:"size:100" json:"nameEn"`
	SubtitleEn         string         `gorm:"size:200" json:"-"`
	ListPrice          float64        `gorm:"column:listPrice;type:decimal(10,2);default:0" json:"listPrice"`
	RewardPoints       int            `gorm:"column:rewardPoints;default:0" json:"rewardPoints"`
	BadgeEn            string         `gorm:"size:20" json:"badgeEn"`
	DescriptionEn      string         `gorm:"type:text" json:"descriptionEn"`
	IconURLEn          string         `gorm:"column:iconUrlEn;size:500" json:"iconUrlEn"`
	IconURLThumbEn     string         `gorm:"column:iconUrlThumbEn;size:500" json:"iconUrlThumbEn"`
	CoverImageEn       string         `gorm:"column:coverImageEn;size:500" json:"coverImageEn"`
	CoverImageThumbEn  string         `gorm:"column:coverImageThumbEn;size:500" json:"coverImageThumbEn"`
	EmojiEn            string         `gorm:"size:20" json:"emojiEn"`
	GradientEn         string         `gorm:"size:300" json:"gradientEn"`
	Status             string         `gorm:"type:enum('active','inactive');default:active" json:"status"`
	CreatedAt          time.Time      `gorm:"column:createdAt" json:"createdAt"`
	UpdatedAt          time.Time      `gorm:"column:updatedAt" json:"updatedAt"`
	Category           *ProductCategory `gorm:"foreignKey:CategoryID" json:"category,omitempty"`
}

func (DeviceGuide) TableName() string { return "device_guides" }
